### ISSUE TITLE
MovingAverage cannot be used with callbacks.ModelCheckpoint

### ISSUE THREAD

**Issue by wuliytTaotao** [tensorflow/addons#903]
**Describe the feature and the current behavior/state.**
Mostly, I save the model at the point of highest val_accuracy by `callbacks.ModelCheckpoint`, but the `MovingAverage` optimizer doesn't seem to support `callbacks.ModelCheckpoint`, i.e., we cannot save the model with moving average in the middle of training processing!

I think we should add MovingAverage into callbacks, and save the moving average model in the middle of training.

**Relevant information**
- Are you willing to contribute it (yes/no):
- Are you willing to maintain it going forward? (yes/no):
- Is there a relevant academic paper? (if so, where):
- Is there already an implementation in another framework? (if so, where): https://github.com/alno/kaggle-allstate-claims-severity/blob/master/keras_util.py
- Was it part of tf.contrib? (if so, where):

**Which API type would this fall under (layer, metric, optimizer, etc.)**
callbacks

**Who will benefit with this feature?**
all tf.keras users

**Any other info.**

**Comment by WindQAQ** [comment]
cc @Squadrick

**Comment by Squadrick** [comment]
@wuliytTaotao Could you post the code that caused the error? I've been trying it out on `tfa-nightly`, and I can't find any error.

**Comment by wuliytTaotao** [comment]
@Squadrick Here is an example, using the model.fit() and callbacks.ModelCheckpoint to save the weights at the point of best validation accuracy, but moving average cannot be added into this processing. I think it's best to implement this function as a callback function.
```
import tensorflow as tf
import os
from tensorflow_addons.optimizers.moving_average import MovingAverage

os.environ["CUDA_VISIBLE_DEVICES"] = '1'

gpus = tf.config.experimental.list_physical_devices(device_type='GPU')
for gpu in gpus:
    tf.config.experimental.set_memory_growth(gpu, enable=True)
print(gpus)

cifar10 = tf.keras.datasets.cifar10

(x_train, y_train), (x_test, y_test) = cifar10.load_data()
x_train, x_test = x_train / 255.0, x_test / 255.0

model = tf.keras.models.Sequential([
    tf.keras.layers.Conv2D(16, (3, 3), padding='same', activation='relu', input_shape=(32, 32, 3)),
    tf.keras.layers.AveragePooling2D(),
    tf.keras.layers.Conv2D(32, (3, 3), padding='same', activation='relu'),
    tf.keras.layers.AveragePooling2D(),
    tf.keras.layers.Flatten(),
    tf.keras.layers.Dense(10, activation='softmax')
])


optimizer = tf.keras.optimizers.Adam(learning_rate=1e-3)
optimizer = MovingAverage(optimizer)

model.compile(optimizer=optimizer,
              loss='sparse_categorical_crossentropy',
              metrics=['accuracy'])

model.fit(x_train, y_train, epochs=5, validation_split=0.1,
          # weights.h5 cannot be saved with moving average,
          # for "average_op" function of MovingAverage cannot be applied during training
          callbacks=[tf.keras.callbacks.ModelCheckpoint('weights.h5', monitor="val_accuracy",
                                                        save_weights_only=True, save_best_only=True)])

model.load_weights("weights.h5")
model.evaluate(x_test, y_test, verbose=2)
```
